docs/i2c.rst

```diff
@@ -21,6 +21,17 @@ should be described separately in that device's documentation.
 Functions
 =========
 
+.. py:function:: init(freq=None, sda=None, scl=None)
+
+    Re-initialize peripheral with the specified frequency ``freq`` on the
+    specified ``sda`` and ``scl`` pins. If pins are not specified, ``pin20``
+    and ``pin19`` are used. If frequency is not specified, it's not changed.
+
+    .. warning::
+
+        Changing the I²C pins from defaults will make the accelerometer and
+        compass stop working, as they are connected internally to those pins.
```

Comment:
I went for the simplest change and tried to make it consistent with `spi.init` and `uart.init`.

Having two i2c's could be nice, but I can't think of an api for this that would both be clear and stay compatible with what is there so far (and with other languages on the micro:bit).